### ISSUE TITLE
Store multiple request body (json) objects

### ISSUE THREAD

**Issue by bkcummins** [a85/POSTMan-Chrome-Extension#9612]
It would be very helpful to be able to store multiple request body (json) objects for a saved endpoint so that you could toggle between them for test data.  

There is a similar option in AWS Lambda to store incoming test events. Thx in advance!
https://aws.amazon.com/blogs/compute/improved-testing-on-the-aws-lambda-console/

I accomplish this now by saving the same endpoint configuration multiple times with different request bodies, but a drop down interface would be ideal.

Thx in advance for consideration! 

Shared originally on the community forum.
https://community.postman.com/t/store-multiple-request-body-json-objects/20315

**Comment by lflucasferreira** [comment]
Maybe this would work like saved examples. In this case, we could create multiple examples and use them in the body request.
I work with an API with a lot of conditional fields and it is very annoying to create multiple requests for the same endpoint.

**Comment by bkcummins** [comment]
Super basic screenshot, but here's a rough idea.

![01](https://user-images.githubusercontent.com/6819212/107981128-84907c00-6f8f-11eb-9602-6cc7642859e3.jpg)

**Comment by bkcummins** [comment]
Super basic screenshot, but here's a rough idea.

![01](https://user-images.githubusercontent.com/6819212/107981203-a853c200-6f8f-11eb-8039-c7e31a9a90ac.jpg)

**Comment by knvpk** [comment]
At first i thought adding example will solve the problem, but examples can't be run as normal request, for now i need to copy paste from example to request and run it. Multiple requests for same url would be very usefull.

**Comment by thatbrainiac** [comment]
Any news on when this may be implemented? :)

**Comment by bridavis** [comment]
This would be extremely helpfull.

**Comment by RaviAH** [comment]
I am waiting for this feature.. any update on this:-)?

**Comment by webjunkie** [comment]
> At first i thought adding example will solve the problem, but examples can't be run as normal request, for now i need to copy paste from example to request and run it. Multiple requests for same url would be very usefull.

Maybe examples could be made "sendable" and possibly "request-only". That would solve it.

**Comment by dinuvld** [comment]
Indeed this would be a very useful addition. We are looking at automatically generating monitors from an OpenAPI spec that has multiple example bodies per request. Being able to natively support this instead of creating multiple requests using the Collection SDK would be an incredible time saver.

**Comment by ewilliams-zoot** [comment]
I haven't actually implemented this myself, but as a work-around, I could see using Collection variables for this. In one variable, you store an index. In another variable you can store a JSON string that has an array of your test cases - called `testCases` - like so:
```json
{
  "testCases": [
    "{\"name\": \"Bob\"}",
    "{\"name\": \"Alice\"}"
  ]
}
```
Then in a pre-request script for the Request, you can do something like this, maybe? Changing the index variable when you want to switch to a different body.
```js
const testCases = JSON.parse(pm.collectionVariables.get("testCases")).testCases;

pm.request.body.raw = testCases[pm.collectionVariables.get("index")];
```
Edit: added `.testCases` to the end of the first JS line

**Comment by blackwiz4rd** [comment]
This feature would be awsome as state above to run automated testing with different bodies.

**Comment by MrWolfPST** [comment]
This is a must have feature.
It should permit store not only the Body section but also Headers and Params, included Query Params and Path Variables.

**Comment by yatw** [comment]
@ewilliams-zoot this work around is hard to maintain in practice. Each API request json string is not going to be nicely display in 1 line and editing something in it is error prone.

**Comment by ewilliams-zoot** [comment]
@yatw Yes, I agree 100%, but it's a work around that seems to have worked for me. I have since tried it on a project, and it is cumbersome to be sure.

**Comment by Buyukcaglar** [comment]
Please bring this functionality to Postman, it really would make our lives a lot easier. 
Thanks.
Onur

**Comment by sksingtn** [comment]
This feature would really help.

**Comment by tylerthecoder** [comment]
I would like this feature a lot. Any update?

**Comment by ewilliams-zoot** [comment]
@tylerthecoder They introduced a CLI with their latest updates; I haven't looked into it yet, but that may have the capability people in this thread are wanting. You can at least integrate whatever functionality they have in the CLI with a scripting language of your choice. I'm not sure that is better than my other suggestion above, to be honest.

**Comment by talesid5** [comment]
I specially googled for this feature that led me to this thread... Would be an amazingly helpful feature

**Comment by mathematikoi** [comment]
this would be a great feature :D

**Comment by g14wx** [comment]
It will be super helpful, sometimes we need to leave different scenarios with different JSON Objects

**Comment by kanchansapkota27** [comment]
Still looking forward for this feature

**Comment by giridharvc7** [comment]
You can save these payloads as `Examples` to document these different bodies and you can consume these trying the examples out. Attaching a short video on how i'd approach this. Let us know if this makes sense. 

https://user-images.githubusercontent.com/1040583/221112612-ab608460-0896-47ae-8c0c-f2f62c62bcfe.mov

**Comment by Buyukcaglar** [comment]
> You can save these payloads as `Examples` to document these different bodies and you can consume these trying the examples out. Attaching a short video on how i'd approach this. Let us know if this makes sense.

This is somewhat convoluted way of achieving the requested functionality but it works. Thanks for sharing. Though I'd prefer directly saving and using multiple request bodies. Until it is implemented properly, this is the way to go.

**Comment by giridharvc7** [comment]
Examples serve as scenarios and documenting what the API is capable of, and thus seems like the perfect place to store multiple json objects for a given request. This way its also easier for your collaborators to understand. Also looking from the previous comments, it seems like one of the blockers all these days was the inability to send examples - this was addressed recently in a release. 
Would love to know your use cases in depth to see if this needs a different solution.

**Comment by webjunkie** [comment]
Linking this issue (sendable examples): https://github.com/postmanlabs/postman-app-support/issues/3694#issuecomment-1433299057

**Comment by rpedroni** [comment]
We use examples for this and works perfectly for showcasing different payloads an endpoint can accept

**Comment by gregg-cbs** [comment]
In 2026 i think this would be useful.

You have a request, you want to toggle the body payload so you can send different payloads.

Makes sense and I was looking for it in Postman today and i was surprised its not a thing. Furthermore im surprised someone wanted this in 2021 and its been neglected.

This will be really useful for companies like Revolut etc who give you postman collections for their APIs where certain endpoints can receive minimal data or comprehensive data.

Creating multiple requests for the same endpoint in postman is not the answer to this and i have seen it done in certain collections and it makes a confusing mess.

I saw it being tabs that you can add, label and remove. Same as how the "Docs", "Params", "Authorisation" tabs look:
<img width="848" height="660" alt="Image" src="https://github.com/user-attachments/assets/42e124a6-0deb-4f3e-af57-dbf5683218ab" />
